### PR TITLE
Adding functionalities

### PR DESCRIPTION
**Pull request by amalexandrescu**
Adding other functionalities on frontend side using readOnly and smartContract functions

### CODE REVIEW

File: src/components/appMenuSections/dashboard/Dashboard.tsx [hunk 1]
```diff
@@ -48,43 +97,70 @@ const Dashboard = () => {
...
+        </ul>
+        {currentRole !== 'Miner' && !clickedJoinPoolButtonByViewer && (
+          <Button
+            variant="contained"
+            onClick={() => {
+              if (currentRole === 'Viewer') {
+                handleJoinPoolButtonByViewer();
+              } else if (currentRole === 'NormalUser' || 'Miner') {
+                ContractAskToJoin(`${userAddress}`);
+                // handleJoinPoolButtonByNormalUser();
```
Comment (Web3RustDev) [inline src/components/appMenuSections/dashboard/Dashboard.tsx:150]: Remove this function call and the implementation of it.

File: src/components/appMenuSections/miningPool/MiningPool.tsx [hunk 2]
```diff
@@ -3,119 +3,56 @@ import TableCell from '@mui/material/TableCell';
...
-  const [finalStatus, setFinalStatus] = useState<string>();
-  const [blocksLeftUntilJoin, setBlocksLeftUntilJoin] = useState<number>();
-  const waitingRows = GetWaitingRows();
   const minersRows = GetMinersRows();
+  const [finalStatus, setFinalStatus] = useState<string>('');
+  const userSession = useAppSelector(selectUserSessionState);
   // will add later, after read_length too big is solved
   // const removalRows = GetRemovalsRows();
 
+  // you'll need this in another file to get the status of the user
```
Comment (Web3RustDev) [inline src/components/appMenuSections/miningPool/MiningPool.tsx]: This is no longer needed

File: src/components/appMenuSections/dashboard/Dashboard.tsx (hunk 1 above)
Comment (BowTiedDeployer) [reply src/components/appMenuSections/dashboard/Dashboard.tsx:150]: it is resolved in a later commit from another branch. has the pr open.

File: src/components/appMenuSections/profile/CommonInfoProfile.tsx [hunk 3]
```diff
@@ -9,10 +9,28 @@ import { network, getExplorerUrl } from '../../../consts/network';
...
+  const [userAddress, setUserAddress] = useState<string | null>(null);
+  const [explorerLink, setExplorerLink] = useState<string | undefined>(undefined);
   const { currentTheme } = useCurrentTheme();
-  const explorerLink = getExplorerUrl[network]().explorerUrl;
+  // const explorerLink = getExplorerUrl[network](userAddress).explorerUrl;
   const userSession = useAppSelector(selectUserSessionState);
 
+  useEffect(() => {
+    if (userSession.isUserSignedIn()) {
+      const args = userSession.loadUserData().profile.stxAddress.testnet;
```
Comment (BowTiedDeployer) [inline src/components/appMenuSections/profile/CommonInfoProfile.tsx:20]: it should be modular testnet/mainnt 

tesnet for devnet/testnet
mainnet for mainnet

File: src/components/appMenuSections/profile/CommonInfoProfile.tsx [hunk 4]
```diff
@@ -9,10 +9,28 @@ import { network, getExplorerUrl } from '../../../consts/network';
...
+  }, [userAddress]);
+
+  useEffect(() => {
+    if (userAddress !== null) {
+      setExplorerLink(getExplorerUrl[network](userAddress).explorerUrl);
+    }
+  }, [explorerLink, userAddress]);
+
   useEffect(() => {
     const wallet = userSession.loadUserData().profile.stxAddress.testnet;
```
Comment (BowTiedDeployer) [inline src/components/appMenuSections/profile/CommonInfoProfile.tsx:35]: same here

> it should be modular testnet/mainnt
> 
> tesnet for devnet/testnet
> mainnet for mainnet

**Review by BowTiedDeployer — DISMISSED** [review]

File: src/components/appMenuSections/profile/CommonInfoProfile.tsx (hunk 3 above)
Comment (Alexandrescu) [reply src/components/appMenuSections/profile/CommonInfoProfile.tsx:20]: https://github.com/stacks-degens/starters-front-end/issues/46

**Review by Alexandrescu — APPROVED** [review]

**Pull request merged**